src/client/Client.cc

```diff
@@ -10097,8 +10097,10 @@ int Client::pwritev(int fd, const struct iovec *iov, int iovcnt, int64_t offset)
 
 int64_t Client::_preadv_pwritev_locked(Fh *fh, const struct iovec *iov,
 				   unsigned iovcnt, int64_t offset, bool write,
-				   bool clamp_to_int, std::unique_lock<ceph::mutex> &cl)
+				   bool clamp_to_int)
 {
+    std::unique_lock cl(client_lock);
+
```

Comment:
Nice cleanup. I have to wonder why this ever took a mutex argument in the first place.

Reply:
The new PR: https://github.com/ceph/ceph/pull/40204